frame/dex-router/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+//! # DEX Router Pallet
+//!
+//! Is used to add route to DEX for given asset_id's pair.
+
+#![cfg_attr(not(test), warn(clippy::disallowed_method, clippy::indexing_slicing))] // allow in tests
+#![warn(clippy::unseparated_literal_suffix, clippy::disallowed_type)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use codec::{Codec, FullCodec};
+	use composable_traits::{
+		defi::LiftedFixedBalance,
+		dex::{CurveAmm, DexRoute, DexRouteNode, DexRouter},
+	};
+	use core::fmt::Debug;
+	use frame_support::pallet_prelude::*;
+	use sp_runtime::{
+		traits::{
+			AtLeast32BitUnsigned, CheckedAdd, CheckedMul, CheckedSub, IntegerSquareRoot, One, Zero,
+		},
+		DispatchResult, FixedPointOperand,
+	};
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type AssetId: FullCodec
+			+ Eq
+			+ PartialEq
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ Debug
+			+ Default
+			+ TypeInfo
+			+ Ord;
+		type Balance: Default
+			+ Parameter
+			+ Codec
+			+ Copy
+			+ Ord
+			+ CheckedAdd
+			+ CheckedSub
+			+ CheckedMul
+			+ AtLeast32BitUnsigned
+			+ From<u64> // at least 64 bit
+			+ Zero
+			+ One
+			+ IntegerSquareRoot
+			+ FixedPointOperand
+			+ Into<LiftedFixedBalance>
+			+ Into<u128>; // cannot do From<u128>, until LiftedFixedBalance integer part is larger than 128
+			  // bit
+		/// The maximum hops in the route.
+		type MaxHopsInRoute: Get<u32> + TypeInfo;
```

Comment:
do we really can ever have u32 hopes? i think u8 would be better. Also, please consider making it #[pallet::constat] so we know it does not chages during runtime. 

i would even make it u4 (via Validatable trat) to make more predictable wights and more resonable amounts of hops. (16 hops is already very big loop)

Reply:
IIUC `BoundedVec` requires `Get<u32>`

Reply:
u8 as u32. so in runtime setup it would be clearly visible the limit. @hussein-aitlahcen would it be helpful to setup runtime if constants will tell what preferred bounds they have? like u8 or u32? i know there are already u4 imitations?